package.json

```diff
@@ -975,30 +975,28 @@
         {
           "id": "librarydataprovider",
           "name": "%views.SAS.libraries%",
-          "when": "(SAS.authorized && SAS.connectionType == rest) || SAS.connection.direct"
+          "when": "((SAS.authorized || SAS.librariesDisplayed) && SAS.librariesEnabled) || SAS.connection.direct"
         },
         {
           "id": "sas-content-get-started",
           "name": "%views.SAS.signIn%",
-          "when": "(!SAS.authorized && SAS.connectionType == rest) && !SAS.connection.direct"
+          "when": "(!(SAS.authorized || SAS.librariesDisplayed) && SAS.canSignIn) && !SAS.connection.direct"
         },
         {
           "id": "sas-content-invalid-connection",
           "name": "%views.SAS.signIn%",
-          "when": "!SAS.authorized && SAS.connectionType != rest"
+          "when": "!SAS.authorized && !SAS.canSignIn"
         }
       ]
     },
     "viewsWelcome": [
       {
         "view": "sas-content-get-started",
-        "when": "(!SAS.authorized && SAS.connectionType == rest) && !SAS.connection.direct",
```

Comment:
im a little lost, why totally delete this? direct and SSH connection still don't need this, right?

Reply:
It looks like these are unneeded because determing whether or not things are displayed is handled in `views.sas-view-container`.

My guess is that the when clauses here can be used to further differentiate what view to show, but I don't think we need it.